hal/network/ncp/cellular/network_config_db.cpp

```diff
@@ -28,23 +29,46 @@ namespace {
 struct NetworkConfig {
     const char* mcc;
     const char* mnc;
+    const char *iccid_prefix;
     const char* apn;
     const char* user;
     const char* password;
 };
 
 const NetworkConfig NETWORK_CONFIG[] = {
-    { "214", "07", "spark.telefonica.com", "", "" }, // Telefonica
-    { "310", "410", "10569.mcs", "", "" }, // Kore/AT&T
-    { "204", "04", "vfd1.korem2m.com", "kore", "kore" } // Kore/Vodafone
+    { "214", "07", "", "spark.telefonica.com", "", "" }, // Telefonica
+    { "310", "410", "", "10569.mcs", "", "" }, // Kore/AT&T
+    { "204", "04", "", "vfd1.korem2m.com", "kore", "kore" } // Kore/Vodafone
+};
+
+const NetworkConfig NETWORK_CONFIG_ICCID[] = {
+    { "", "", "8988307", "super", "", "" }, // Twilio
+    { "", "", "8988323", "super", "", "" }  // Twilio
 };
 
 const size_t NETWORK_CONFIG_SIZE = sizeof(NETWORK_CONFIG) / sizeof(NETWORK_CONFIG[0]);
+const size_t NETWORK_CONFIG_ICCID_SIZE = sizeof(NETWORK_CONFIG_ICCID) / sizeof(NETWORK_CONFIG_ICCID[0]);
 
 const size_t MCC_SIZE = 3;
+const size_t ICCID_MIN_SIZE = 19;
+const size_t ICCID_PREFIX_LEN = 7;
 
 } // unnamed
 
+CellularNetworkConfig networkConfigForIccid(const char* iccid, size_t size) {
+    if (size < ICCID_MIN_SIZE + 7) {    // 7 bytes for "+CCID: " in AT response
```

Comment:
We should definitely not pass raw AT response here.

Reply:
Fixed.